web/libs/datamanager/src/stores/Tabs/store.js

```diff
@@ -174,7 +174,14 @@ export const TabStore = types
         yield getRoot(self).apiCall("deleteTab", { tabID: view.id });
       }
 
-      destroy(view);
+      if (view.virtual && newView) {
+        // this helps resolve the race condition between the view being deleted and the new view being selected
+        setTimeout(() => {
+          self.deleteView(view, { autoselect: false });
+        }, 0);
```

Comment:
Could we not just put the destroy always to be on the next tick, it is hard to understand the intent of this with a recursive function call.

Reply:
I tried that but it was throwing this error `Cannot modify 'View[]@/viewsStore/views', the object is protected and can only be modified by using an action.`